Source/cmake/OptionsGTK.cmake

```diff
@@ -387,14 +387,15 @@ if (USE_OPENGL_OR_ES)
 
     if (ENABLE_WEBGL)
         # ANGLE-backed WebGL depends on DMABuf support, which at the moment is leveraged
-        # through libgbm and libdrm dependencies. USE_LIBGBM and USE_TEXTURE_MAPPER_DMABUF
-        # also have to be defined in that case.
-
-        find_package(GBM REQUIRED)
-        find_package(LibDRM REQUIRED)
-
-        SET_AND_EXPOSE_TO_BUILD(USE_LIBGBM TRUE)
-        SET_AND_EXPOSE_TO_BUILD(USE_TEXTURE_MAPPER_DMABUF TRUE)
+        # through libgbm and libdrm dependencies. When libgbm is found to be available,
+        # make libdrm a requirement and define the USE_LIBGBM and USE_TEXTURE_MAPPER_DMABUF
+        # macros. When not available, ANGLE will be used in slower software-rasterization mode.
+        find_package(GBM)
+        if (GBM_FOUND)
```

Comment:
I don't like this because now it's easy to fall back to software rasterization by mistake if gbm.pc is not installed. All of the other dependency checks are careful to avoid this (except for the decision to use OpenGL vs. OpenGL ES). To avoid this problem, a new build flag should be introduced to allow building without GBM, and the build should fail if the flag is not passed, to make sure this can only happen intentionally and not by mistake.

Reply:
Alternatively, we could just require it when `ENABLE_WEBGL` is ON. That's currently tied to `USE_OPENGL_OR_ES`.

Reply:
Reported https://bugs.webkit.org/show_bug.cgi?id=247132.